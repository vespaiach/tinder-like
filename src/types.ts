export interface APIResult<T> {
    ok: boolean;
    body?: {
        data: T[];
        total: number;
        page: number;
    };
    error?: string;
}

export interface UserProfile {
    id: string;
    lastName: string;
    firstName: string;
    email: string;
    title: string;
    picture: string;
}

/**
 * Provider a way to get data from server
 */
export type DataProvider = {
    getList(page: number): Promise<APIResult<UserProfile>>;
};
