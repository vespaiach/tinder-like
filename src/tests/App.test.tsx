import React from 'react';
import { render } from '@testing-library/react';

import AppShell from '../App';
import { DataProvider } from '../types';
import body from './data.test.json';
import { act } from 'react-dom/test-utils';

/**
 * Mocking dataProvider
 */
const dataProvider: DataProvider = {
    getList: () => Promise.resolve({ ok: true, body }),
    getOne: (id) =>
        Promise.resolve({
            ok: true,
            body: {
                id,
                phone: '0700-3090279',
                lastName: 'Fiedler',
                firstName: 'Heinz-Georg',
                gender: 'male',
                picture: 'https://randomuser.me/api/portraits/men/81.jpg',
                dateOfBirth: '1974-03-12T21:15:08.878Z',
            },
        }),
};

/**
 * Taking snapshot of app shell
 * Take the link below to see why we need to make test functions asynchronously.
 * https://github.com/facebook/react/issues/14769
 */
describe('Test app shell', () => {
    test('should be on home page', async () => {
        let container;
        await act(async () => {
            const result = render(<AppShell dataProvider={dataProvider} />);
            container = result.container;
        });

        expect(container).toMatchSnapshot();
    });
});
