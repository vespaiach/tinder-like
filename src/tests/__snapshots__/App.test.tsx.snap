// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app shell should be on home page 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-containerRoot-2 MuiContainer-maxWidthXs"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="makeStyles-navigation-1"
        >
          <button
            aria-label="go to home page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-edgeStart"
            data-testid="home-page"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root MuiDivider-vertical"
          />
          <button
            aria-label="list of liked profiles"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            data-testid="liked-page"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.35 20.13c-.76.69-1.93.69-2.69-.01l-.11-.1C5.3 15.27 1.87 12.16 2 8.28c.06-1.7.93-3.33 2.34-4.29 2.64-1.8 5.9-.96 7.66 1.1 1.76-2.06 5.02-2.91 7.66-1.1 1.41.96 2.28 2.59 2.34 4.29.14 3.88-3.3 6.99-8.55 11.76l-.1.09z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root MuiDivider-vertical"
          />
          <button
            aria-label="list of disliked profiles"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            data-testid="disliked-page"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <title>
                  Dislike
                </title>
                <path
                  d="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C8.17,3 8.82,3.12 9.44,3.33L13,9.35L9,14.35L12,21.35V21.35M16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35L11,14.35L15.5,9.35L12.85,4.27C13.87,3.47 15.17,3 16.5,3Z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-appShell-3"
          id="app-shell-container"
          style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Kayla Bredesen"
              class="MuiCardMedia-root makeStyles-img-4 MuiCardMedia-media MuiCardMedia-img"
              src="https://randomuser.me/api/portraits/women/13.jpg"
              title="Kayla Bredesen"
            />
            <div
              class="MuiCardContent-root makeStyles-cardContent-5"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Kayla Bredesen
              </h2>
              <div
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                Sex: 
                male
                <br />
                Age: 
                47
              </div>
            </div>
            <div
              class="MuiCardActions-root makeStyles-footerRoot-6 MuiCardActions-spacing"
            >
              <div
                class="MuiButtonGroup-root MuiButtonGroup-fullWidth MuiButtonGroup-disableElevation MuiButtonGroup-contained"
                role="group"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation MuiButton-fullWidth"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <span
                      class="MuiButton-startIcon MuiButton-iconSizeSmall"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <title>
                          Broken Heart
                        </title>
                        <path
                          d="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C8.17,3 8.82,3.12 9.44,3.33L13,9.35L9,14.35L12,21.35V21.35M16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35L11,14.35L15.5,9.35L12.85,4.27C13.87,3.47 15.17,3 16.5,3Z"
                        />
                      </svg>
                    </span>
                    Dislike
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation MuiButton-fullWidth"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <span
                      class="MuiButton-startIcon MuiButton-iconSizeSmall"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M13.35 20.13c-.76.69-1.93.69-2.69-.01l-.11-.1C5.3 15.27 1.87 12.16 2 8.28c.06-1.7.93-3.33 2.34-4.29 2.64-1.8 5.9-.96 7.66 1.1 1.76-2.06 5.02-2.91 7.66-1.1 1.41.96 2.28 2.59 2.34 4.29.14 3.88-3.3 6.99-8.55 11.76l-.1.09z"
                        />
                      </svg>
                    </span>
                    Like
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
